include/common.h

```diff
@@ -23,18 +23,15 @@ void inspect_pointer(void *ptr)
 
 void error(char* string) {
 	fputs(string, stderr);
+	fputc('\n', stderr);
 }
 
-// This should be a macro or inlined
-// In the current case it gives the stack pointer
-// inside of the cheri_csp_get function.
-// This is fine if only the capabilitits are important.
-// In any other case the csp will need to be decremented
-// to get the actual csp of the calling function.
-void* cheri_csp_get()
+
+// This function returns the current stack pointer.
+//
+// We don't want to inline it because __builtin_frame_address will change it's meaning.
```

Comment:
I think this line needs expanding a bit. I *think* I know what it's suggesting, but I reckon we need another line or so of explanation to make sure that other idiots (for example, me) can understand what's going on :)

Reply:
I hope this explanation is sufficient.
https://github.com/capablevms/cheri-examples/blob/fdcb452a0716ac7f85acac01bf8ebf24ad54d659/include/common.h#L30-L34

Reply:
Works for me!